src/v/pandaproxy/rest/handlers.cc

```diff
@@ -82,27 +85,31 @@ get_brokers(server::request_t rq, server::reply_t rp) {
         return kafka::metadata_request{.list_all_topics = false};
     };
     return rq.service()
-      .client()
-      .local()
-      .dispatch(make_metadata_req)
-      .then([res_fmt, rp = std::move(rp)](
-              kafka::metadata_request::api_type::response_type res) mutable {
-          json::get_brokers_res brokers;
-          brokers.ids.reserve(res.data.brokers.size());
-
-          std::transform(
-            res.data.brokers.begin(),
-            res.data.brokers.end(),
-            std::back_inserter(brokers.ids),
-            [](const kafka::metadata_response::broker& b) {
-                return b.node_id;
-            });
-
-          auto json_rslt = ppj::rjson_serialize(brokers);
-          rp.rep->write_body("json", json_rslt);
-
-          rp.mime_type = res_fmt;
-          return std::move(rp);
+      .client_cache()
+      .fetch_client(rq.ctx.user, rq.ctx.authn_type)
+      .then([res_fmt, &make_metadata_req, rp = std::move(rp)](
+              client_ptr client) mutable {
+          return client->real.dispatch(make_metadata_req)
+            .then(
+              [res_fmt, rp = std::move(rp)](
+                kafka::metadata_request::api_type::response_type res) mutable {
+                  json::get_brokers_res brokers;
+                  brokers.ids.reserve(res.data.brokers.size());
+
+                  std::transform(
+                    res.data.brokers.begin(),
+                    res.data.brokers.end(),
+                    std::back_inserter(brokers.ids),
+                    [](const kafka::metadata_response::broker& b) {
+                        return b.node_id;
+                    });
+
+                  auto json_rslt = ppj::rjson_serialize(brokers);
+                  rp.rep->write_body("json", json_rslt);
+
+                  rp.mime_type = res_fmt;
+                  return std::move(rp);
+              });
```

Comment:
I think this can be unnested one:
```suggestion
      .client_cache()
      .fetch_client(rq.ctx.user, rq.ctx.authn_type)
      .then([&make_metadata_req](client_ptr client) mutable {
          return client->real.dispatch(make_metadata_req);
      })
      .then([res_fmt, rp = std::move(rp)](
              kafka::metadata_request::api_type::response_type res) mutable {
          json::get_brokers_res brokers;
          brokers.ids.reserve(res.data.brokers.size());

          std::transform(
            res.data.brokers.begin(),
            res.data.brokers.end(),
            std::back_inserter(brokers.ids),
            [](const kafka::metadata_response::broker& b) {
                return b.node_id;
            });

          auto json_rslt = ppj::rjson_serialize(brokers);
          rp.rep->write_body("json", json_rslt);

          rp.mime_type = res_fmt;
          return std::move(rp);
```

Reply:
Excellent suggestion.